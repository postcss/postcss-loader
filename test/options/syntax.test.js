const { webpack } = require('@webpack-utilities/test')

describe('Options', () => {
  test('Syntax - {String}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          syntax: 'sugarss'
        }
      }
    }

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1]

      expect(source).toEqual('module.exports = "a\\n  color: black\\n"')
      expect(source).toMatchSnapshot()
    })
  })

  test('Syntax - {Object}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          ident: 'postcss',
          syntax: require('sugarss')
        }
      }
    }

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1]

      expect(source).toEqual('module.exports = "a\\n  color: black\\n"')
      expect(source).toMatchSnapshot()
    })
  })
})
