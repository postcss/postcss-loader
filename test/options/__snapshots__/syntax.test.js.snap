// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Syntax should work Syntax - {Object}: css 1`] = `
"a
  color: black
"
`;

exports[`Options Syntax should work Syntax - {Object}: errors 1`] = `Array []`;

exports[`Options Syntax should work Syntax - {Object}: warnings 1`] = `Array []`;

exports[`Options Syntax should work Syntax - {String}: css 1`] = `
"a
  color: black
"
`;

exports[`Options Syntax should work Syntax - {String}: errors 1`] = `Array []`;

exports[`Options Syntax should work Syntax - {String}: warnings 1`] = `Array []`;
